### PR TITLE
Final library modularization for 2.11

### PR DESCRIPTION
**Pull request by adriaanm**
The continuations plugin and library moved to https://github.com/scala/scala-continuations.
The Scala Swing library moved to https://github.com/scala/scala-swing.

They will still ship with 2.11 (albeit unsupported), but may disappear in 2.12.

TODO: add deprecation to CPS plugin over at https://github.com/scala/scala-continuations.

review by @gkossakowski

NOTE: I've moved the java7 patches over to scala/scala-swing as well, which means we're one step closer to building master on java 6 and 7 without modification. I noticed I missed a couple of changes to partest javaagent in the java7 branch. It would be good to port those over, and to see if we can get rid of the java7 branch all together in scala/scala.

### CODE REVIEW

**Comment by adriaanm** [comment]
Another TODO on my list: submit https://github.com/adriaanm/scala-ide/compare/scala-ide:master...modularize as a PR

**Comment by retronym** [comment]
One general point that occurred to me: the neg tests in modules are tightly coupled to the compiler error message formatting. We should refactor them to avoid duplicating that assumption I'm each test. That will give us one place per module to adapt when those messages evolve.

**Comment by adriaanm** [comment]
Agreed. There's already some decoupling going by only looking at substrings, but it could be improved further.

**Comment by skyluc** [comment]
Scala IDE is packaging the continuation plugin, and using it to test enabling compiler plugins for the presentation compiler and the sbt builder. So we need to update our references to it.

**Comment by gkossakowski** [comment]
@skyluc: if continuations plugin is used just for testing basic integration with plugin infrastructure then would it make sense to write a very simple one just for that purpose and get rid of dependency on continuations for that purpose?

**Comment by skyluc** [comment]
No, we added it because it is (was?) part of the Scala distribution. The idea being that you have access to the same content when using the Scala distribution or Scala IDE. Then we started to use it to check that configuring compiler plugins is working fine.
But already diverge from the Scala distribution, because we don't package akka.actor.

@dragos, any thoughts on this?

**Comment by adriaanm** [comment]
Please see my first comment -- I'm already preparing a PR:  https://github.com/adriaanm/scala-ide/compare/scala-ide:master...modularize

**Comment by adriaanm** [comment]
In general, I'd prefer it if the IDE would fall back when module jars aren't available -- it should work with just lib/reflect/comp. We'll still package the CPS plugin with the 2.11 dist, but we're not fixing bugs in it.

**Comment by gkossakowski** [comment]
FYI: I haven't forgot to review this PR. I'd like to understand a few more things about it. I'll have a look during the weekend.

**Comment by adriaanm** [comment]
(last update bumps the new module versions to RC2, now with less sbt-osgi induced madness)

**Comment by adriaanm** [comment]
The failures should be fixed by the IDE PR I sent.

**Comment by gkossakowski** [comment]
You missed editing other Eclipse and IntelliJ project files:

```
(pull/3221)$ grep -R 'continuations' src/eclipse/ src/intellij/
src/eclipse//reflect/.classpath:    <classpathentry combineaccessrules="false" kind="src" path="/continuations-library"/>
src/eclipse//scala-compiler/.classpath: <classpathentry combineaccessrules="false" exported="true" kind="src" path="/continuations-library"/>
src/eclipse//test-junit/.classpath: <classpathentry combineaccessrules="false" kind="src" path="/continuations-library"/>
src/intellij//scala-lang.ipr.SAMPLE:      <module fileurl="file://$PROJECT_DIR$/continuations-library.iml" filepath="$PROJECT_DIR$/continuations-library.iml" />
src/intellij//scala-lang.ipr.SAMPLE:      <module fileurl="file://$PROJECT_DIR$/continuations-plugin.iml" filepath="$PROJECT_DIR$/continuations-plugin.iml" />
```

The same goes for swing. Also, does `src/build/dbuild-meta-json-gen.scala` need to be updated?

Other than that this looks good to me assuming PR validation failures are addressed.

**Comment by adriaanm** [comment]
Yep! You're right. I'll add a commit for that as soon as the other IDE issues are fixed.

**Comment by adriaanm** [comment]
Retrying failed part of validation: https://scala-webapps.epfl.ch/jenkins/job/pr-scala-integrate-ide/4746/console

**Comment by adriaanm** [comment]
For the record, validating changes that require coordinating PRs between scala/scala and scala-ide/scala-ide, is much easier now that we have the `scriptsRemote` parameter in the pr-scala-\* jobs and enabled the rebuild plugin (but could be easier with a bit more jenkins script hacking -- would be nice to be able to just tweak any parameter used in the scripts without forking any repos). 

For now, to test validation against your own fork of scala/scala-ide, fork scala/jenkins-scripts, and patch it [thusly](https://github.com/adriaanm/jenkins-scripts/commit/e4aa10067ff2dc643ca86a1f0f51bae3cd7def6e), fork scala-ide/uber-build and [tweak it juuust a smidge](https://github.com/adriaanm/uber-build/commit/04b616619e23fa4cdcc9b1c98eb58f612ca1024c), then rebuild the failed part of the pr-scala flow, pointing the `scriptsRemote` parameter to your fork. /cc @dotta, @huitseeker, @skyluc

File: build.xml [hunk 1]
```diff
@@ -107,6 +107,10 @@ TODO:
   <target name="distclean"   depends="dist.clean"   description="Removes all distributions. Binaries and documentation are untouched."/>
 
 
+  <target name="test.continuations.suite">
+    <echo message="DEPRECATED: the continuations have moved to https://github.com/scala/scala-continuations"/>
```
Comment (soc) [inline build.xml:111]: s/the continuations have/the continuations plugin has/ or s/the continuations have/continuations have/?

Comment (adriaanm) [inline build.xml:111]: I appreciate your attention to detail, but I promise you this line will disappear as soon as I get a chance to update scala/jenkins-scripts.

**Comment by adriaanm** [comment]
IDE/dbuild configs cleaned up

**Pull request merged**